mne/gui/__init__.py

```diff
@@ -247,6 +247,97 @@ def locate_ieeg(info, trans, aligned_ct, subject=None, subjects_dir=None,
     return gui
 
 
+@verbose
+def view_vol_stc(stcs, freq_first=True, subject=None, subjects_dir=None,
+                 src=None, inst=None, show_topomap=True, show=True,
+                 block=False, verbose=None):
+    """View a volume time and/or frequency source time course estimate.
+
+    Parameters
+    ----------
+    stcs : list of list | generator
+        The source estimates, the options are: 1) List of lists or
+        generators for epochs and frequencies (i.e. using
+        :func:`mne.minimum_norm.apply_inverse_tfr_epochs` or
+        :func:`mne.beamformer.apply_dics_tfr_epochs`-- in this case
+        use ``freq_first=False``), or 2) List of source estimates across
+        frequencies (e.g. :func::func:`mne.beamformer.apply_dics_csd`),
+        or 3) list of source estimates across epochs
+        (e.g. :func:`mne.minimum_norm.apply_inverse_epochs` and
+        :func:`mne.beamformer.apply_dics_epochs`--in this
+        case use ``freq_first=False``), or 4) Single
+        source estimates (e.g. :func:`mne.minimum_norm.apply_inverse`
+        and :func:`mne.beamformer.apply_dics`, note ``freq_first``
+        will not be used in this case).
+    freq_first : bool
+        If frequencies are the outer list of ``stcs`` use ``True``.
+    %(subject)s
+    %(subjects_dir)s
+    src : instance of SourceSpaces
+        The volume source space for the ``stc``.
+    inst : EpochsTFR | AverageTFR | None
+        The time-frequency or data object to use to plot topography.
+    show_topomap : bool
+        Whether to show the sensor topomap in the GUI.
+    show : bool
+        Show the GUI if True.
+    block : bool
+        Whether to halt program execution until the figure is closed.
+    %(verbose)s
+
+    Returns
+    -------
+    gui : instance of VolSourceEstimateViewer
+        The graphical user interface (GUI) window.
+    """
+    from ..viz.backends._utils import _init_mne_qtapp, _qt_app_exec
+    from ._vol_stc import VolSourceEstimateViewer, COMPLEX_DTYPE, RANGE_VALUE
+
+    app = _init_mne_qtapp()
+
+    # cast to integers to lower memory usage, use custom complex data
+    # type if necessary
+    data = list()
+    # can be generator, compute using first stc object, just a general
+    # rescaling of data, does not need to be precise
+    scalar = None
+    for inner_stcs in (stcs if np.iterable(stcs) else [stcs]):
+        inner_data = list()
+        for stc in (inner_stcs if np.iterable(inner_stcs) else [inner_stcs]):
+            if np.iscomplexobj(stc.data):
+                if scalar is None:
+                    # this is an order of magnitude approximation,
+                    # larger stcs will have some clipping
+                    scalar = (RANGE_VALUE - 1) / stc.data.real.max() / 5
```

Comment:
So since it's can be a generator, we can't loop through twice so we need to scale the first time. The assumption is that the data in one stc isn't more than 5 times another stc. Maybe this should be a bit more conservative though